packages/fxa-settings/src/pages/InlineTotpSetup/index.test.tsx

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { screen, render, fireEvent, waitFor } from '@testing-library/react';
+// import { getFtlBundle, testL10n } from 'fxa-react/lib/test-utils';
+// import { FluentBundle } from '@fluent/bundle';
+import InlineTotpSetup from '.';
+import { logPageViewEvent } from '../../lib/metrics';
+import { MozServices } from '../../lib/types';
+import { MOCK_CODE } from './mocks';
+
+jest.mock('../../lib/metrics', () => ({
+  logViewEvent: jest.fn(),
+  logPageViewEvent: jest.fn(),
+}));
+
+describe('InlineTotpSetup', () => {
+  // let bundle: FluentBundle;
+  beforeAll(async () => {
+    global.URL.createObjectURL = jest.fn();
+    //   bundle = await getFtlBundle('settings');
+  });
+  it('renders default as expected', () => {
+    render(<InlineTotpSetup code={MOCK_CODE} />);
+    // const ftlMsgMock = screen.getAllByTestId('ftlmsg-mock')[1];
+    // testL10n(ftlMsgMock, bundle, {
+    //   email: exampleEmail,
+    // });
+    screen.getByRole('heading', {
+      name: `Enable two-step authentication to continue to ${MozServices.Default}`,
+    });
+    expect(
+      screen.getByLabelText('A device with a hidden 6-digit code.')
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getByRole('button', { name: 'Continue' })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'Cancel setup' })
+    ).toBeInTheDocument();
+  });
+
+  it('renders intro view as expected with custom service name', () => {
+    render(
+      <InlineTotpSetup
+        code={MOCK_CODE}
+        serviceName={MozServices.FirefoxMonitor}
+      />
+    );
+    // const ftlMsgMock = screen.getAllByTestId('ftlmsg-mock')[1];
+    // testL10n(ftlMsgMock, bundle, {
+    //   email: exampleEmail,
+    // });
+    screen.getByRole('heading', {
+      name: `Enable two-step authentication to continue to ${MozServices.FirefoxMonitor}`,
+    });
+    expect(
+      screen.getByLabelText('A device with a hidden 6-digit code.')
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getByRole('button', { name: 'Continue' })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'Cancel setup' })
+    ).toBeInTheDocument();
+  });
+
+  it('renders QR code by default when a user clicks "Continue"', async () => {
+    render(<InlineTotpSetup code={MOCK_CODE} />);
+    // const ftlMsgMock = screen.getAllByTestId('ftlmsg-mock')[1];
+    // testL10n(ftlMsgMock, bundle, {
+    //   email: exampleEmail,
+    // });
+    const continueButton = screen.getByRole('button', { name: 'Continue' });
+    fireEvent.click(continueButton);
+    await waitFor(() =>
```

Comment:
Do we need `await waitFor(() =>`? See how I did this in `cookies_disabled` tests 🤔